pkg/performanceprofile/cmd/render/render.go

```diff
@@ -191,45 +194,101 @@ func render(inputDir, outputDir string) error {
 			return err
 		}
 
-		uid := pp.UID
-		if uid == types.UID("") {
-			uid = uuid.NewUUID()
-		}
-
-		or := []v1.OwnerReference{
-			{
-				Kind:       pp.Kind,
-				Name:       pp.Name,
-				APIVersion: pp.APIVersion,
-				UID:        uid,
-			},
-		}
-
-		for _, componentObj := range components.ToObjects() {
-			componentObj.SetOwnerReferences(or)
+		if addOwnerRef {
+			err = addOwnerReference(components, pp)
+			if err != nil {
+				return err
+			}
 		}
-
-		for kind, manifest := range components.ToManifestTable() {
-			b, err := yaml.Marshal(manifest)
+		if len(labelWeakRef) > 0 {
+			addWeakOwnerReferenceLabel(components, pp, labelWeakRef)
+			err = writeObject(outputDir, fmt.Sprintf("%s_%s.yaml", pp.Name, "annotated"), pp)
 			if err != nil {
 				return err
 			}
+		}
 
-			fileName := fmt.Sprintf("%s_%s.yaml", pp.Name, strings.ToLower(kind))
-			fullFilePath := filepath.Join(outputDir, fileName)
-			klog.Info("Writing file: ", fullFilePath)
-
-			err = os.WriteFile(fullFilePath, b, 0644)
+		for kind, manifest := range components.ToManifestTable() {
+			err = writeObject(outputDir, fmt.Sprintf("%s_%s.yaml", pp.Name, strings.ToLower(kind)), manifest)
 			if err != nil {
 				return err
 			}
-			klog.Info(fileName)
 		}
 	}
 
 	return nil
 }
 
+func writeObject(outputDir, fileName string, manifest interface{}) error {
+	fullFilePath := filepath.Join(outputDir, fileName)
+	klog.Infof("Writing file: %s -> %s", fileName, fullFilePath)
+
+	b, err := yaml.Marshal(manifest)
+	if err != nil {
+		return err
+	}
+
+	return os.WriteFile(fullFilePath, b, 0644)
+}
+
+func addOwnerReference(components *manifestset.ManifestResultSet, pp *performancev2.PerformanceProfile) error {
+	if pp == nil || pp.UID == types.UID("") {
+		return fmt.Errorf("Missing UID from performance profile")
+	}
+
+	or := []v1.OwnerReference{
+		{
+			Kind:       pp.Kind,
+			Name:       pp.Name,
+			APIVersion: pp.APIVersion,
+			UID:        pp.UID,
+		},
+	}
+
+	for _, componentObj := range components.ToObjects() {
+		componentObj.SetOwnerReferences(or)
+	}
+
+	return nil
+}
+
+const (
+	weakOwnerReferenceLabel = "performance.openshift.io/weak-owner-reference"
+)
+
+func addWeakOwnerReferenceLabel(components *manifestset.ManifestResultSet, pp *performancev2.PerformanceProfile, labelWeakRef string) {
+	ref := makeWeakOwnerRef(labelWeakRef, pp)
+
+	klog.Infof("using weak ref label %q -> %q", labelWeakRef, ref)
+
+	if pp.Labels == nil {
+		pp.Labels = make(map[string]string)
+	}
+	pp.Labels[weakOwnerReferenceLabel] = ref
```

Comment:
How will the relink implementation later distinguish name from uuid? It can just try, that is fine, but lets make sure the plan is documented.

Reply:
thing is, we can't (easily/cleanly) because we mix types in the label values: can either be a string (name) or a uuid, thuse the field we use of a key depends.

I think the way forward is to either always use the name or the uuid. Both have their drawbacks.

Reply:
or to use different labels for name vs uuid?

Reply:
Why do we want to give the option to choose between name and uid?

I thought the "weak reference" was a way to rebuild the owner ref structure as we cannot do it directly in the rendered objects, so what we use to do it seems like an implementation detail to me. All we need is a unique identifier shared by both the PP and the rendered objects so we could rebuild the "link" in the controller, right?

Reply:
> Why do we want to give the option to choose between name and uid?

because uid is the strongest we can achieve, but is not predictable; we can make predictable, but then it becomes hard to consume (because we need to mutate the performance profile and send to the cluster the _mutated_ performance profile).
Name is predictable, so we're considering which one we should use.

> I thought the "weak reference" was a way to rebuild the owner ref structure as we cannot do it directly in the rendered objects, so what we use to do it seems like an implementation detail to me. All we need is a unique identifier shared by both the PP and the rendered objects so we could rebuild the "link" in the controller, right?

Yes, that's correct, this is the intent.